sdk/go/common/workspace/templates_zip.go

```diff
@@ -154,10 +163,18 @@ func RetrieveZIPTemplateFolder(templateURL *url.URL, tempDir string) (string, er
 		return "", err
 	}
 	packageRequest.Header.Set("Accept", "application/zip")
+
+	for _, opt := range opts {
+		opt(packageRequest)
+	}
+
 	packageResponse, err := client.Do(packageRequest)
 	if err != nil {
 		return "", err
 	}
+	if packageResponse.StatusCode == http.StatusUnauthorized && isPulumiHostResponse(packageResponse) {
```

Comment:
Does pulumi cloud return a useful error in this case that we could print to the user, similar to https://github.com/pulumi/pulumi/pull/16825/files#diff-7fa5262aff46d0920b6cc69bd3b500dec2fac9ba8069932d7d422990c3396f39R183?  I'm guessing not really, and the error message below is fine?

Also, should we give users a hint of how they can resolve this issue, e.g. by login into pulumi cloud?